pycloudlib/azure/cloud.py

```diff
@@ -782,6 +793,88 @@ def _check_for_network_interfaces(self):
 
         return None
 
+    def _get_resource_sku_info(self, instance_type: str) -> Tuple[str, int]:
+        """Return the family and vCPU count for a VM size in a given region.
+
+        Raises:
+            ValueError: the specified VM size was not found or vCPU capability is invalid.
+        """
+        resource_sku_list = self.compute_client.resource_skus.list(
+            filter=f"location eq '{self.location}'"
+        )
+
+        if not resource_sku_list:
+            raise ValueError(
```

Comment:
In order to conform with https://github.com/canonical/pycloudlib/issues/263, this and the rest of exceptions in this file should be pycloudlib-specific, to facilitate catching them in call-sites.